test/svid/jwtsvid/test_jwt_svid.py

```diff
@@ -172,6 +176,117 @@ def test_valid_parse_insecure(test_input_token, test_input_audience, expected):
 """
     parse_and_validate tests
 
-    TBD
-
 """
+
+
+@pytest.mark.parametrize(
+    'test_input_token,test_input_jwtBundle, test_input_audience, expected',
+    [
+        (
+            '',
+            None,
+            ['spire'],
+            INVALID_INPUT_ERROR.format('token cannot be empty'),
+        ),
+        (
+            'eyJhbGciOiJFUzI1NiIsImtpZCI6Imd1eTdsOWZSQzhkQW1IUmFtaFpQbktRa3lId2FHQzR0IiwidHlwIjoiSldUIn0.eyJhdWQiOlsib3RoZXItc2VydmljZSJdLCJleHAiOjE2MTIyOTAxODMsImlhdCI6MTYxMjI4OTg4Mywic3ViIjoic3hthrtmZlOi8vZXhhbXBsZS5vcmcvc2VydmljZSJ9.W7CLQvYVBQ8Zg3ELcuB1K9hE4I9wyCMB_8PJTZXbjnlMBcgd0VDbSm5OjoqcGQF975eaVl_AdkryJ_lzxsEQ4A',
+            None,
+            ['spire'],
+            INVALID_INPUT_ERROR.format('jwt_bundle cannot be empty'),
+        ),
+    ],
+)
+def test_invalid_parameters_parse_and_validate(
+    test_input_token, test_input_jwtBundle, test_input_audience, expected
+):
+    with pytest.raises(ValueError) as exception:
+        JwtSvid.parse_and_validate(
+            test_input_token, test_input_jwtBundle, test_input_audience
+        )
+    assert str(exception.value) == expected
+
+
+def test_valid_parse_and_validate_RSA():
+    rsa_key = rsa.generate_private_key(public_exponent=65537, key_size=2048)
+    jwt_bundle = JwtBundle(TrustDomain('test.orgcom'), {'kid1': rsa_key.public_key()})
+    rsakeypem, _ = get_keys_pems(rsa_key)
+    token = jwt.encode(
+        {
+            'aud': ['spire', 'test', 'valid'],
+            'sub': 'spiffe://test.orgcom/',
+            'exp': timegm(
+                (
+                    datetime.datetime.utcnow() + datetime.timedelta(hours=1)
+                ).utctimetuple()
+            ),
+        },
+        algorithm="RS256",
+        key=rsakeypem,
+        headers={'alg': 'RS256', 'typ': 'JWT', 'kid': 'kid1'},
+    )
+    JwtSvid.parse_and_validate(token, jwt_bundle, 'spire')
+    assert True
```

Comment:
As `parse_and_validate` returns a jwt_svid, we could check that the returned object is correct.  (Same for next test)

Reply:
great idea!